utils/products.ts

```diff
@@ -1,19 +1,53 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import {ProductComponent} from '../types/store/plugins';
+import {useLocation, matchPath} from 'react-router-dom';
 
-export const getCurrentProductId = (products: ProductComponent[], pathname: string): string | null => {
-    if (!products) {
-        return null;
-    }
+import {useSelector} from 'react-redux';
 
-    for (let i = 0; i < products.length; i++) {
-        const product = products[i];
-        if (pathname.startsWith(product.baseURL)) {
-            return product.id;
-        }
-    }
+import {ProductComponent} from 'types/store/plugins';
+import {selectProducts, selectCurrentProductId, selectCurrentProduct} from 'selectors/products';
```

Comment:
Are we creating a circular dependency here since `selectCurrentProduct{Id}` depend on `getCurrentProduct`?

BTW, do we need `getCurrentProduct{Id}`?

Reply:
Indeed there are a lot of cross-references between these two modules, but no the individual methods shouldn't create any circular dependencies. Flow: `use` -> `select` -> `get`.

When I refactored these, I made sure there were `get`, `select`, and `use` varieties for both `ProductComponent` and `ProductIdentifier`. These are used in a lot of different contexts (connected, functional components, etc.) and I felt the flexibility was worth supporting even though it's not currently used.

That said, I'm 0/5 on leaving `getCurrentProductId` in or not—doesn't matter to me.